README.md

```diff
@@ -104,17 +169,81 @@ const client = await connect({
   vhost: "/",
 })
 
-const consumerOptions = { stream: "stream-name", offset: Offset.next() }  // see docs for various offset types
+const consumerOptions = { stream: "stream-name", offset: Offset.next() } // see docs for various offset types
 
 const consumer = await client.declareConsumer(consumerOptions, (message: Message) => {
-  console.log(message.content) // it's a Buffer 
+  console.log(message.content) // it's a Buffer
+})
+
+// declareConsumer works even with batch publishing and compression
+
+// ...
+
+await client.close()
+```
+
+### Single Active Consumer
+
+It is possible to create a consumer as single active.
+For the given consumer reference only him will be able to consume messages
+
+```typescript
+const consumerOptions = {
+  stream: "stream-name",
+  offset: Offset.next(),
+  singleActive: true,
+  consumerRef: "my-consumer-ref",
+} // see docs for various offset types
+
+const consumer = await client.declareConsumer(consumerOptions, (message: Message) => {
+  console.log(message.content) // it's a Buffer
+})
+// ...
+```
+
+### Clustering
+
+Every time we create a new producer or a new consumer, a new connection is created.
+In particular for the producer the connection is created on the node leader.
+For more running the tests in a cluster follow the readme under the folder /cluster
+
+### Load Balancer
+
+With the load balancer, what happens is we will connect first to the AddressResolver
+and then we will connect to the node through the AddressResolver.
+The address resolver is going to give us a node leader for a Producer and a node
+replica for the consumer, otherwise it will close the connection and retry.
+
+```typescript
+const firstNode = "node0:5562" // can be taken from envs
+const resolver = getAddressResolverFromEnv()
+const client = await connect({
+  hostname: "node0",
+  port: 5562,
+  username: "rabbit",
+  password: "rabbit",
+  vhost: "/",
+  addressResolver: { enabled: true, endpoint: resolver },
```

Comment:
here we have addressResolver whit enabled and endpoint @tarzacodes can we make more ergonomic using hostname and port from the outside and collapsing it in just a boolean?

Reply:
Are you sure? I think that it'd be quite limiting if we were to force people to put their resolver in the env vars.... maybe they want to relate it to some other endpoint or get it from the network or any other number of options. Might be reasonable, if enabled is true and no endpoint is specified, to read the env vars as a default, WDYT?

Reply:
Ok, I made it so that if the endpoint is not defined it'll use the client hostname/port